Lib/fontTools/ttLib/tables/_g_l_y_f.py

```diff
@@ -1012,33 +1012,37 @@ def getCoordinates(self, glyfTable):
 					coordinates, endPts, flags = g.getCoordinates(glyfTable)
 				except RecursionError:
 					raise ttLib.TTLibError("glyph '%s' contains a recursive component reference" % compo.glyphName)
+				coordinates = GlyphCoordinates(coordinates)
 				if hasattr(compo, "firstPt"):
-					# move according to two reference points
+					# component uses two reference points: we apply the transform _before_
+					# computing the offset between the points
+					if hasattr(compo, "transform"):
+						coordinates.transform(compo.transform)
 					x1,y1 = allCoords[compo.firstPt]
 					x2,y2 = coordinates[compo.secondPt]
 					move = x1-x2, y1-y2
-				else:
-					move = compo.x, compo.y
-
-				coordinates = GlyphCoordinates(coordinates)
-				if not hasattr(compo, "transform"):
 					coordinates.translate(move)
 				else:
-					apple_way = compo.flags & SCALED_COMPONENT_OFFSET
-					ms_way = compo.flags & UNSCALED_COMPONENT_OFFSET
```

Comment:
the whole scaled vs unscaled component _offset_ (Apple vs MS) logic should not apply for components that use firstPt/secondPt anchor points. There is no offset to scale in this case; the only offset possible is the difference between firstPt and secondPt, which must always be aligned to one another (whether the componet is scaled or not).